app/models/concerns/user_multifactor_methods.rb

```diff
@@ -54,6 +54,7 @@ def mfa_required_weak_level_enabled?
     end
 
     def ui_otp_verified?(otp)
+      return false if mfa_seed.blank?
```

Comment:
Is this even possible? If a user has webauthn enabled, but not TOTP, do previous recovery codes actually do anything? From my understanding the recovery codes are only valid if TOTP is enabled, in which case `mfa_seed` would not be blank. I don't believe recovery codes will do anything if TOTP is disabled?

Reply:
Fixed. I ran into an issue with that where `verify_digit_otp` is called when mfa is disabled, in order to enable it, [see here](https://github.com/Shopify/RubyGems.org/blob/5c80da6f4a0bde0d489954a592fc902a52ff43c7/app/models/concerns/user_multifactor_methods.rb#L21). So I just had to check if the seed passed to it was also blank in addition to `mfa_seed`. This is because for webauthn both of these would be blank, but when attempting to enable mfa `mfa_seed` would be blank, but `seed` wouldn't be.

Reply:
Oh 🤦 I think you're right. Fixed